pertpy/tools/_perturbation_space/_perturbation_space.py

```diff
@@ -364,50 +365,57 @@ def label_transfer(
         self,
         adata: AnnData,
         column: str = "perturbation",
+        column_uncertainty_score_key: str = "perturbation_transfer_uncertainty",
         target_val: str = "unknown",
-        n_neighbors: int = 5,
-        use_rep: str = "X_umap",
+        neighbors_key: str = "neighbors",
     ) -> None:
         """Impute missing values in the specified column using KNN imputation in the space defined by `use_rep`.
 
         Args:
             adata: The AnnData object containing single-cell data.
-            column: The column name in AnnData object to perform imputation on.
+            column: The column name in adata.obs to perform imputation on.
+            column_certainty_score: The column name in adata.obs to store the certainty score of the label transfer.
             target_val: The target value to impute.
-            n_neighbors: Number of neighbors to use for imputation.
-            use_rep: The key in `adata.obsm` where the embedding (UMAP, PCA, etc.) is stored.
+            neighbors_key: The key in adata.uns where the neighbors are stored.
 
         Examples:
             >>> import pertpy as pt
             >>> import scanpy as sc
             >>> import numpy as np
             >>> adata = sc.datasets.pbmc68k_reduced()
-            >>> rng = np.random.default_rng()
-            >>> adata.obs["perturbation"] = rng.choice(
-            ...     ["A", "B", "C", "unknown"], size=adata.n_obs, p=[0.33, 0.33, 0.33, 0.01]
-            ... )
+            >>> # randomly dropout 10% of the data annotations
+            >>> adata.obs["perturbation"] = adata.obs["louvain"].astype(str).copy()
+            >>> random_cells = np.random.choice(adata.obs.index, int(adata.obs.shape[0] * 0.1), replace=False)
+            >>> adata.obs.loc[random_cells, "perturbation"] = "unknown"
             >>> sc.pp.neighbors(adata)
             >>> sc.tl.umap(adata)
             >>> ps = pt.tl.PseudobulkSpace()
-            >>> ps.label_transfer(adata, n_neighbors=5, use_rep="X_umap")
+            >>> ps.label_transfer(adata)
         """
-        if use_rep not in adata.obsm:
-            raise ValueError(f"Representation {use_rep} not found in the AnnData object.")
-
-        embedding = adata.obsm[use_rep]
-
-        from pynndescent import NNDescent
-
-        nnd = NNDescent(embedding, n_neighbors=n_neighbors)
-        indices, _ = nnd.query(embedding, k=n_neighbors)
-
-        perturbations = np.array(adata.obs[column])
-        missing_mask = perturbations == target_val
+        if neighbors_key not in adata.uns:
```

Comment:
Why are you checking `uns`? I think it's obsp! See returns section here:

https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pp.neighbors.html

Reply:
See previous comment.